### PR TITLE
colored call diff (basic)

### PR DESCRIPTION
**Pull request by FeepingCreature**
feature: when call doesn't match with existing expectation with the same name, provide colored diff description highlighting the failure

Instead of showing "group expectation: not fulfilled X, extra call Y" match up missing call and extra call and try to provide more detailed diff.

ping @linkrope 

TODO unittests

### CODE REVIEW

File: source/dmocks/repository.d [hunk 1]
```diff
@@ -183,16 +184,67 @@ class MockRepository
...
+                if (callExpectation.satisfied) return;
+                foreach (i, unexpectedCall; unexpectedCalls)
+                {
+                    if (callExpectation.name.matches(unexpectedCall.name))
+                    {
+                        // name matches: assume they are meant to match up, generate parameter diff
+                        expectationError ~= "\n";
+                        expectationError ~= CallExpectationDiff(unexpectedCall, callExpectation);
+                        expectation = null;
+                        unexpectedCalls = unexpectedCalls.remove(i);
```
Comment (belka-ew) [inline source/dmocks/repository.d]: Is it safe to remove elements during the iteration?

I have an alternative solution that may work, but I'm not insisting it is better, but it also eliminates the `if` in the `foreach` (it filters the elements before the loop).
```d
import std.range : dropBack;

if (callExpectation.satisfied)
{
    return;
}
alias pred = call => callExpectation.name.matches(call.name);
auto matched = unexpectedCalls.partition!(pred, SwapStrategy.stable);

foreach (unexpectedCall; unexpectedCalls.dropBack(matched.length))
{
    // name matches: assume they are meant to match up, generate parameter diff
    expectationError ~= "\n";
    expectationError ~= CallExpectationDiff(unexpectedCall, callExpectation);
    expectation = null;
}
unexpectedCalls = matched;
return;
```

File: source/dmocks/expectation.d [hunk 2]
```diff
@@ -68,6 +68,31 @@ class CallExpectation : Expectation
...
+    {
+        string result;
+
+        if (object != call.object)
+        {
+            result ~= yellow("(different object).");
+        }
+        if (!name.matches(call.name))
+        {
+            assert(false); // then why consider us an attempted match?
```
[the remark this answers is not in the document]
Comment (FeepingCreature) [reply source/dmocks/expectation.d:81]: done

File: source/dmocks/repository.d [hunk 3]
```diff
@@ -183,16 +184,67 @@ class MockRepository
...
+                    }
+                }
+                return;
+            }
+            assert(false); // what kind of expectation is this?!
```
[the remark this answers is not in the document]
Comment (FeepingCreature) [reply source/dmocks/repository.d:231]: done

File: source/dmocks/util.d [hunk 4]
```diff
@@ -59,3 +59,26 @@ public class MocksSetupException : Exception {
...
+    return text;
+}
+
+private bool enableAnsiEscape()
+{
+    version (Posix)
+    {
+        import core.sys.posix.unistd : isatty;
+
+        return isatty(1) ? true : false; // stdout
```
[the remark this answers is not in the document]
Comment (FeepingCreature) [reply source/dmocks/util.d:78]: `int` kann nicht automatisch nach `bool` konvertiert werden.

File: source/dmocks/repository.d (hunk 1 above)
Comment (FeepingCreature) [reply source/dmocks/repository.d]: habs so ähnlich gemacht

**Pull request merged**